specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -2071,86 +3243,249 @@
+    "EndpointDeploymentResourcePropertiesBasicResourceArmPaginatedResult": {
       "type": "object",
       "properties": {
+        "nextLink": {
+          "type": "string"
+        },
         "value": {
-          "$ref": "#/definitions/DiagnoseRequestProperties"
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+          }
         }
       }
     },
-    "EncryptionKeyVaultUpdateProperties": {
-      "required": [
-        "keyIdentifier"
-      ],
+    "EndpointKeys": {
       "type": "object",
       "properties": {
-        "keyIdentifier": {
-          "minLength": 1,
-          "pattern": "[a-zA-Z0-9_]",
+        "keys": {
+          "description": "Dictionary of Keys for the endpoint.",
+          "type": "object",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "EndpointModels": {
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The link to the next page constructed using the continuationToken.  If null, there are no additional pages.",
           "type": "string"
+        },
+        "value": {
+          "description": "List of models.",
+          "type": "array",
+          "items": {}
         }
       }
     },
-    "EncryptionProperty": {
+    "EndpointResourceProperties": {
       "required": [
-        "keyVaultProperties",
-        "status"
+        "endpointType"
       ],
       "type": "object",
       "properties": {
-        "cosmosDbResourceId": {
-          "description": "The byok cosmosdb account that customer brings to store customer's data\r\nwith encryption",
+        "associatedResourceId": {
+          "description": "Byo resource id for creating the built-in model service endpoints.",
           "type": "string"
         },
-        "identity": {
-          "description": "Identity to be used with the keyVault",
-          "$ref": "#/definitions/IdentityForCmk"
-        },
-        "keyVaultProperties": {
-          "description": "KeyVault details to do the encryption",
-          "$ref": "#/definitions/KeyVaultProperties"
+        "endpointType": {
+          "description": "Type of the endpoint.",
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
         },
-        "searchAccountResourceId": {
-          "description": "The byok search account that customer brings to store customer's data\r\nwith encryption",
+        "endpointUri": {
+          "description": "Uri of the endpoint.",
           "type": "string"
         },
-        "status": {
-          "$ref": "#/definitions/EncryptionStatus"
+        "failureReason": {
+          "description": "The failure reason if the creation failed.",
+          "type": "string"
         },
-        "storageAccountResourceId": {
-          "description": "The byok storage account that customer brings to store customer's data\r\nwith encryption",
+        "name": {
```

Comment:
>name [](http://example.com/codeflow?start=9&length=4)

Rename

Reply:
This  API is already in use and calling by UX, which is hard to change the name, is it possible that we keep this implementation for now?

Reply:
This is a new definitions in a new API version, so it's not released to customers or even widely internally is it? Now is the time to change it. This isn't a critical blocking issue, but definitely recommended.